lib/modules/datasource/rubygems/get.ts

```diff
@@ -182,4 +200,87 @@ export class InternalRubyGemsDatasource extends Datasource {
 
     return new Marshal(response.body).parsed as T;
   }
+
+  private async isNexusDataSource(registry: string): Promise<boolean> {
+    const statusEndPoint = '/service/rest/v1/status';
+    const endPointRe = regEx(`^(?<endPoint>.*:\\d{1,5})\\/`);
+    const { endPoint } = registry.match(endPointRe)?.groups ?? {
+      endPoint: '',
+    };
+    const nexusUrl = endPoint.concat(statusEndPoint);
+    let isNexus = false;
+    try {
+      const response = await this.http.getJson<HttpResponse>(nexusUrl);
+      isNexus = response.headers?.server?.includes('Nexus') ?? false;
+    } catch (err) {
+      logger.debug({ err }, 'Not a nexus datasource');
+    }
+    return isNexus;
+  }
+
+  private async getReleasesFromNexus(
+    registry: string,
+    dependencyName: string
+  ): Promise<Release[]> {
+    const result: Release[] = [];
+    try {
+      const url = this.getNexusGemReleasesEndPoint(registry, dependencyName);
+      const nexusResponse = (await this.http.getJson<NexusGems>(url))?.body || {
+        continuationToken: null,
+        items: [],
+      };
+      const gemItems: JsonNexusGemsItems[] = [...nexusResponse.items];
+
+      let continuationToken: string | null = nexusResponse.continuationToken;
+      while (continuationToken) {
+        const gemsOfPage: NexusGems = await this.pageOfNexusGems(
+          url,
+          continuationToken
+        );
+        gemItems.push(...gemsOfPage.items.filter(Boolean));
+        continuationToken = gemsOfPage.continuationToken;
+      }
+      result.push(
+        ...gemItems.map((item) => {
+          return { version: item.version, rubyPlatform: 'nexus' };
+        })
+      );
+    } catch (error) {
+      logger.debug({ error }, 'Failed to retreive gems from nexus');
+      throw error;
+    }
+    return result;
+  }
+
+  private getNexusGemReleasesEndPoint(
+    registry: string,
+    dependencyName: string
+  ): string {
+    const registryArray = registry.split('/');
+    const repository = registryArray[registryArray.indexOf('repository') + 1];
```

Comment:
doing split by `/` then `indexOf`, i believe it's better to use a regex, and get the repository in your group, 
correct me if i am wrong since i dont know the contents of 'registry'